src/test/validator.test.ts

```diff
@@ -62,62 +49,64 @@ describe('FormValidator', () => {
+            confirmPasswordInput.name = 'Confirm Password';
+
+            passwordInput.value = 'password123';
+            confirmPasswordInput.value = 'password123';
             // when 
-            const passwordsMatch = formValidator.arePasswordsMatching();
+            const result = formValidator.arePasswordsMatching();
             // then
-            expect(passwordsMatch).toBe(false);
+            expect(result).toEqual(true);
         });
     })
 
     describe('Error paragraph', () => {
         it('create error paragraph', () => {
             // given
-            const validationError = 'Invalid email address.';
-            const inputElement = inputCreator;
+            const errorParagraph = new HtmlTagName().paragraphCreator();
+            errorParagraph.id = 'Email-error'
+            errorParagraph.textContent = 'Invalid email address.'
+
+            const inputElement = document.createElement('input');
             inputElement.name = 'Email';
+            const inputElementNode = document.createElement('div');
+            inputElementNode.appendChild(inputElement);
+            document.body.appendChild(inputElementNode);
+            inputElementNode.insertBefore(errorParagraph, inputElement.nextSibling);
+
             // when
-            const errorParagraph = formValidator.createErrorParagraph(validationError, inputElement);
-        
+            const error = formValidator.createErrorParagraph(errorParagraph.textContent, inputElement);
+
             // then
-            expect(errorParagraph).toBeTruthy();
+            expect(error).toBeTruthy();
         });
+
         it('clear existing error paragraph', () => {
-            const inputElement = inputCreator;
-            inputElement.name = 'Email';
-            formValidator.clearExistingError(inputElement);
-            const usernameErrorParagraph = document.getElementById('email-error');
-            expect(usernameErrorParagraph).toBeNull();
-        });
-    })
+            // given
+            const errorParagraph = new HtmlTagName().paragraphCreator();
+            errorParagraph.id = 'Email-error';
+            document.body.appendChild(errorParagraph);
 
-    describe('validateForm', () => {
-        it('return true if a form is valid', () => {
-            const inputElements = document.querySelectorAll('input');
-    
-            const isValidMock = jest.fn().mockReturnValue(true);
-        
-            const result = new FormValidator().validateForm(inputElements);
-        
-            expect(result).toBe(true);
-            expect(isValidMock).toHaveBeenCalledTimes(inputElements.length);
+            // when
+            const removeParagraph = formValidator.removeErrorParagraph(errorParagraph);
+
+            // then
+            expect(removeParagraph).toBeUndefined();
         });
-    })
+    });
+
+    it('validateForm', () => {
+        // given
+        const inputElements = document.querySelectorAll('input');
```

Comment:
gdzie jest ten input definiowany w tym dokumencie? Ogolnie testy powinny byc bezstanowe. Czyli powinnas kontext testow budowac tu na biezaco chyba ze moze byc ogolny dla wszystkich testow w danym pliku to wtedy nie trzeba budowac per test. Ale pod warunkiem ze ten kontekst sie nie zmienia. Czyli ze nie jest tak ze najpierw Tworzysz formualrz a drugi test tworzy inputy a 3 je sprawdza itd bo wtedy kazdy test zalezy od wyniku poprzedniego wskutek czego testy staja sei bezuzyteczne